src/Package/Impl/Options/R/Tools/RSettingsImplementation.cs

```diff
@@ -126,6 +128,16 @@ internal sealed class RSettingsImplementation : BindableBase, IRSettings {
             set => SetProperty(ref _lastActiveConnection, value);
         }
 
+        public string LastLocalDockerUsername {
+            get => _lastLocalDockerUsername;
+            set => SetProperty(ref _lastLocalDockerUsername, value);
+        }
+
+        public string LastLocalDockerPassword {
```

Comment:
This appears to be per VS session as it is not persisted in the Tools | Options. Expected?

Reply:
If it does need to be persisted it may need to be instead stored in the Windows cred storage.

Reply:
https://github.com/Microsoft/RTVS/issues/3939
`it is not persisted in the Tools | Options` - it expected to be similar to `LastActiveConnection`. Did I miss something?

Reply:
Won't be encrypted...

Reply:
Ok, we need some place to store it. For #3939, I wanted to save credentials using container name (which is also a connection name) as an authority. I can create another authority (something like "RTVS:LastDockerCredentials") and get username from it. But I'll have to convert password into normal string to pass it to the docker.

Reply:
https://technet.microsoft.com/library/jj554668.aspx

I believe we have `NativeMethods` for it